docs/3.0-migration-guide.d/relaxed-psk-semantics.md

```diff
@@ -0,0 +1,23 @@
+Relaxed semantics for PSK configuration
+-----------------------------------------------------------------
+
+This affects users which call the PSK configuration APIs
+`mbedtlsl_ssl_conf_psk()` and `mbedtls_ssl_conf_psk_opaque()`
+multiple times on the same SSL configuration.
+
+In Mbed TLS 2.x, users would observe later calls overwriting
+the effect of earlier calls, with the prevailing PSK being
+the one that has been configured last.
+
+To achieve equivalent functionality when migrating to Mbed TLS 3.0,
+users calling `mbedtls_ssl_conf_[opaque_]psk()` multiple times should
+remove all but the last call, so that only one call to _either_
+`mbedtls_ssl_conf_psk()` _or_ `mbedtls_ssl_conf_psk_opaque()`
+remains.
+
+However, if the _intent_ of the multiple calls to
```

Comment:
I think this is too forward-thinking. This guide is about how to migrate to actual 3.0, not some unspecified 3.x. The current state of affairs is that if you want to provision multiple PSKs, you're out of luck and can only get in touch on the mailing list so that we can prioritize this enhancement.

Reply:
As above, I disagree: If users have multiple PSKs they want to configure, they should call `mbedtls_ssl_conf_psk()` multiple times from 3.0 onwards, even if the PSK limit in 3.0 is still `1`. This way, they don't have to change anything as 3.X evolves, as it should be.

Reply:
My main issue with the current wording is I think it's misleading. If I'm a user and read this entry, the logical conclusion I will draw is that Mbed TLS gained support for provisioning multiple PSKs in a single context, which it has not. Granted, if I then go read the doxygen documentation up to the first note I'll see that it's in fact not the case, but (1) are we sure all users will read that note? and (2) anyway, can't we just avoid giving the wrong impression in the first place.

> This way, they don't have to change anything as 3.X evolves, as it should be.

Reply:
> This way, they don't have to change anything as 3.X evolves, as it should be.

Actually that's not even fully the case, unless we also specify right now what the behaviour is in case multiple keys are passed with the same identity (and perhaps other edge cases we failed to consider so far, but which are likely to become apparent when we get to implementing and testing it). And at this point let's be practical: why push work on ourselves now that we can do later? We already have too much on our plate for 3.0 already.